campaignion_wizard/src/Email.php

```diff
@@ -214,21 +224,137 @@ protected function saveEmail($email_type, &$form, &$form_state) {
     // webform_email_edit_form_submit() sets messages. We don't want them.
     // The simplest way seems to be to backup & restore $_SESSION['messages'].
     $messages = isset($_SESSION['messages']) ? $_SESSION['messages'] : FALSE;
+
     webform_email_edit_form_submit($form, $form_state);
     $_SESSION['messages'] = $messages;
     if (!$_SESSION['messages']) {
       unset($_SESSION['messages']);
     }
 
+    $fields = array('email_type' => $email_type);
+    $fields['request_lifetime'] = $this->calculateLifetime($form_state);
+
+
     db_merge('webform_confirm_email')
       ->key(array(
         'nid' => $this->node->nid,
         'eid' => $this->eid,
       ))
-      ->fields(array('email_type' => $email_type))
+      ->fields($fields)
       ->execute();
   }
 
+  /**
+   * Inserts a field labeled "Advanced Settings" which contains fields for
+   * configuring the request lifetime settings of an email.
+   */
+  protected function insertLifetimeFields($email, &$email_form) {
```

Comment:
We should try to leverage whatever `webform_confirm_email` provides in terms of form elements here.